doc/SAI-Proposal-ECMP-Member-Count.md

```diff
@@ -0,0 +1,41 @@
+# ECMP Count Capability and Configuration
+
+
+This document talks about the 
+- Capability query of MAX ECMP Count supported by silicon
+- Configuration
+- SAI adapter expectation
+
+## Problem Description
+There is a mechanism to set the HW default of max ecmp member count using a Key-Value string SAI_KEY_NUM_ECMP_MEMBERS during the system boot time. This KV string can be set only once and either is based on prior knowledge of HW capability or assumed default. SAI adapter consumes this Key-Value string and sets the ecmp member count in HW. Set value can be queried by a read only attribute SAI_SWITCH_ATTR_ECMP_MEMBERS.
+
+## Spec Enhancement
+Two new Switch attributes are introduced to address the abovementioned issue.
+
+> SAI_SWITCH_ATTR_MAX_ECMP_MEMBER_COUNT (Read Only)
+> This is a capability query during switch init to find out device specific max number of ecmp members supported.

```

Comment:
Please clarify based on our discussion today:

```
This is a capability query during switch init to find out device specific max number of ecmp members supported across all Nexthop Groups.
```

Reply:
captured in comments